pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -528,7 +528,12 @@ func convertRbacRulesToFilterConfig(service *serviceMetadata, option rbacOption)
 			ShadowRules: permissiveRbac}
 	}
 
-	return &http_config.RBAC{Rules: rbac, ShadowRules: permissiveRbac}
+	// If RBAC permissive mode is only set on policy level, set ShadowRules only when there is policy in permissive mode.
```

Comment:
Could you also note what's the result without this patch? i.e. what happens when ShadowRules is set when len(permissiveRbac.Policies) is 0.